### PR TITLE
Fix skeleton test for Vertex AI resources

### PR DESCRIPTION
**Pull request by jingyih**
### Change description

Fixes #

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

**Comment by jingyih** [comment]
@maqiuyujoyce Could you help take a look if this PR makes sense?

File: config/servicemappings/vertexai.yaml [hunk 1]
```diff
@@ -64,7 +64,7 @@ spec:
     - name: google_vertex_ai_endpoint
       kind: VertexAIEndpoint
       idTemplate: "projects/{{project}}/locations/{{region}}/endpoints/{{name}}"
-      idTemplateCanBeUsedToMatchResourceName: false
+      idTemplateCanBeUsedToMatchResourceName: true
```
Comment (maqiuyujoyce) [inline config/servicemappings/vertexai.yaml:67]: Is this change on purpose? This property/field is usually set to true when we want to enable import/export tool for this resource. Or is it configured because we want to use export for vertex AI resources in e2e tests?

File: pkg/resourceskeleton/testdata/uri-skeleton.yaml [hunk 2]
```diff
@@ -1111,34 +1111,30 @@
...
-    apiVersion: vertexai.cnrm.cloud.google.com/v1beta1
-    kind: VertexAIDataset
-    spec:
-      projectRef:
-    external: kcc-test
-    location: us-central1
+- ExpectedSkeleton: null # VertexAIDataset does not support import
   ResourceConfigId: VertexAIDataset
-  URI: "https://us-central1-aiplatform.googleapis.com/v1beta1/projects/kcc-test/locations/us-central1/datasets/1234567890?alt=json"
+  URI: ""
```
Comment (maqiuyujoyce) [inline pkg/resourceskeleton/testdata/uri-skeleton.yaml:1116]: Looking at other examples, URI may or may not be kept when ExpectedSkeleton is null. Do you mind adding a unit test to ensure the behavior is consistent? I think either not setting it (if possible) or setting it to an empty string should work.

File: pkg/resourceskeleton/testdata/uri-skeleton.yaml [hunk 3]
```diff
@@ -1111,34 +1111,30 @@
...
+        external: kcc-test
+      region: us-central1
   ResourceConfigId: VertexAIEndpoint
   URI: "https://us-central1-aiplatform.googleapis.com/v1beta1/projects/kcc-test/locations/us-central1/endpoints/vertexaiendpoint-test"
 - ExpectedSkeleton:
     apiVersion: vertexai.cnrm.cloud.google.com/v1beta1
     kind: VertexAIIndex
-    metadata:
-      name: vertexaiindex-test
+    metadata: {}
```
Comment (maqiuyujoyce) [inline pkg/resourceskeleton/testdata/uri-skeleton.yaml:1131]: Why is the metadata section cleared up?

**Review by maqiuyujoyce — COMMENTED** [review]
LGTM

File: config/servicemappings/vertexai.yaml (hunk 1 above)
Comment (jingyih) [reply config/servicemappings/vertexai.yaml:67]: I am not familiar with this field. But according to #2135, if this field is set to `false`, test `TestNewFromURI` will skip on the resource?

Comment (maqiuyujoyce) [reply config/servicemappings/vertexai.yaml:67]: You are right. And the field was designed for the export/import tool: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/pkg/apis/core/v1alpha1/servicemapping_types.go#L88

File: pkg/resourceskeleton/testdata/uri-skeleton.yaml (hunk 2 above)
Comment (jingyih) [reply pkg/resourceskeleton/testdata/uri-skeleton.yaml:1116]: Related to the previous comment, the test `TestNewFromURI` is currently skipped for `VertexAIDataset`. Do we still need URI?

File: pkg/resourceskeleton/testdata/uri-skeleton.yaml (hunk 3 above)
Comment (jingyih) [reply pkg/resourceskeleton/testdata/uri-skeleton.yaml:1131]: As the result of enabling the test `TestNewFromURI` for resource (by setting `idTemplateCanBeUsedToMatchResourceName` to `true`), this is the expected output of the test. It is an output only field in [API](https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations.endpoints). So it ended up showing up under `status.observedState.name`. Not sure if I miss configured the service mapping?

File: pkg/resourceskeleton/testdata/uri-skeleton.yaml (hunk 2 above)
Comment (maqiuyujoyce) [reply pkg/resourceskeleton/testdata/uri-skeleton.yaml:1116]: I don't think we need it!

Comment (jingyih) [reply pkg/resourceskeleton/testdata/uri-skeleton.yaml:1116]: Just double check on my understanding. I saw that resource `VertexAIDataset` does not support import. Therefore we should set `idTemplateCanBeUsedToMatchResourceName` to `false` for this resource?

https://registry.terraform.io/providers/hashicorp/google-beta/4.84.0/docs/resources/vertex_ai_dataset#import

Comment (maqiuyujoyce) [reply pkg/resourceskeleton/testdata/uri-skeleton.yaml:1116]: In most cases, you are correct. More specifically, the decision flow I have for this field is:

1. Do I need to support export for this resource? Yes: Go to step 2. No: Set the value to `false`.
2. Can the `IDTemplate` be used to match the resource's URL? Yes: Set the value to `true`. No: Go to step 3.
3. Figure out the export use case / design for this resource.

Whether TF support import of this resource or not can be helpful to answer the question in step 2 above.

File: pkg/resourceskeleton/testdata/uri-skeleton.yaml (hunk 3 above)
Comment (jingyih) [reply pkg/resourceskeleton/testdata/uri-skeleton.yaml]: Thanks! I set `idTemplateCanBeUsedToMatchResourceName: false` for VertexAIIndex.

**Review by maqiuyujoyce — COMMENTED** [review]
/lgtm
/approve
